src/main/java/edu/studyup/serviceImpl/EventServiceImpl.java

```diff
@@ -29,25 +29,29 @@ public Event updateEventName(int eventID, String name) throws StudyUpException {
 		return event;
 	}
 
+	//BUG #2: Inefficient use of keySet iterator
+	//SOLUTION: use entrySet iterator
 	@Override
 	public List<Event> getActiveEvents() {
 		Map<Integer, Event> eventData = DataStorage.eventData;
 		List<Event> activeEvents = new ArrayList<>();
 		
-		for (Integer key : eventData.keySet()) {
-			Event ithEvent= eventData.get(key);
+		for (Map.Entry<Integer, Event> key : eventData.entrySet()) {
+			Event ithEvent= key.getValue();
 			activeEvents.add(ithEvent);
 		}
 		return activeEvents;
 	}
 
+	//BUG #3: Inefficient use of keySet iterator
+	//SOLUTION: use entrySet iterator
 	@Override
 	public List<Event> getPastEvents() {
 		Map<Integer, Event> eventData = DataStorage.eventData;
 		List<Event> pastEvents = new ArrayList<>();
 		
-		for (Integer key : eventData.keySet()) {
-			Event ithEvent= eventData.get(key);
+		for (Map.Entry<Integer, Event> key : eventData.entrySet()) {
```

Comment:
Same as BUG#2, I would do the same thing to enhance the efficiency of this for-loop.